README.adoc

```diff
@@ -0,0 +1,290 @@
+This guide uses the MicroProfile OpenAPI as a UI for interacting with the servers.
+
+== Getting started
+include::{common-includes}/gitclone.adoc[]
+
+=== Try what you’ll build
+The `finish` directory in the root of this guide contains the finished application.
+Give it a try before you proceed.
+
+To try out the application, first navigate to the `finish` directory and then run
+the following Maven goal to build the application and run it inside Open Liberty:
+
+```
+cd finish
+mvn liberty:start-server -DserverName=server1
+```
+
+Point your browser to the link:http://localhost:9080/openapi/ui/[] URL. This will
+display the available REST endpoints on the `server1` server.
+
+First, make a POST request to `/sessions/session/cart/{item}&{price}`.
+The POST request will add a user specified item and price to a session
+that represents data in a user's cart.
+
+Then, make a GET request to `/sessions/session/cart`. The GET request
+will return all the items from your session and display it as a string.
+
+When you are ready to continue stop your server and navigate to the `start` directory.
+
+```
+mvn liberty:stop-server -DserverName=server1
+cd ../start
+```
+
+== Learning about sessions
+A `session` is a way to store information to be used across multiple pages.
+
+When you work with a local application (such as a word processor), you open it,
+make changes, and then you close it. The computer knows that you opened the app,
+made changes, and closed the app. On the internet, however, the web server does
+not know who you are, or what you do, because it is just processing stateless HTTP
+requests.
+
+Session variables store user information like usernames, items in a cart, etc.
+By default, session variables will timeout after 30 minutes of being unused.
+Session variables are maintained on a web server. `Cookies`, which also store user
+information, are maintained on a clients computer.
+
+== Learning about session persistence
+The primary use case of `session persistence` is to be able to handle client requests
+being routed to different backends or to handle failover of the backend system.
+
+High traffic websites must support thousands of users in a fast and reliable way.
+Load balancing requires running several instances of the same service in parallel
+so that traffic can be routed to servers to maximize speed and reliability.
+
+This poses a problem when each server keeps an isolated copy of their session data.
+In this case, a user is tied to a particular server, and if that server crashes
+then that server's session data will be lost.
+
+image::sessionCache.png[]
+
+`Session persistence` using a `session cache`, as illustrated above, solves this
+problem by allowing all server instances to share caches amongst each other. This
+eliminates the need to always route a user to the same server instance, and also
+helps in failover situations by distributing the cache.
+
+== Building and running the application
```

Comment:
@JunyiSun The `Building and running the application` section starts on line 95.

Reply:
@KyleAure Hey I realized that your `Building and running the application` does not mention anything about how to compile and run the application, which does not serve the purpose of this section. You can take a look at how our common  `Building and running the application` section does here: https://raw.githubusercontent.com/OpenLiberty/guides-common/master/mvnbuild.adoc

Reply:
We use shared common file in our guides. As an example, in the https://github.com/OpenLiberty/guide-cdi-intro, you can see how mvnbuild.adoc, mvnverify.adoc files get linked there. We should follow this style to make our guides consistent. Thank you.